### DISCUSSION TITLE
Multiple Body(...) Parameters Request Conflict

### DISCUSSION THREAD

**Discussion by woosal1337** [fastapi/fastapi#9546] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
UpdateUserInfo.py


from typing import Optional

from pydantic import BaseModel


class UpdateUserInfo(BaseModel):
    """
    This class is used to create a model for info data.
    """

    name: Optional[str]
    surname: Optional[str]
    email: Optional[str]


main.py
```python
@app.post("/api/v1/update_profile")
async def update_profile(token: str = Header(...),
                         update_user_info: UpdateUserInfo = Body(...),
                         update_baho_info: UpdateUserInfo = Body(...)):
    """
    :param token:
    :param update_user_info:
    :return:
    """
    try:
        user = aj.decode_apple_jwt(token)

        if user:
            username: str = user["sub"]

            dict_user_info = update_user_info.dict()
            print(dict_user_info)

            dict_baho_info = update_baho_info.dict()
            print(dict_baho_info)

            return True

        else:
            return False

    except Exception as e:
        logging.error(e)
        return e
```

POST Request Input:
```python
import requests
import json

url = "http://localhost:8000/api/v1/update_profile"

headers = {
    'Content-Type': 'application/json',
    'token': 'my_token'
}

data = {
    "update_user_info": {
        "name": "new_name",
        "surname": "new_surname",
        "email": "new_email@example.com"
    },
    "update_baho_info": {
        "name": "new_name",
        "surname": "new_surname",
        "email": "baho@gmail.com"
    }
}

response = requests.post(url, headers=headers, data=json.dumps(data))

print(response.status_code)
print(response.json())
```

### Description

With the code above where 2 Body(...) type input alongside the header parameters were set can be easily sent a POST request, which completely works fine with the following input.

```json
{
    "update_user_info": {
        "name": "new_name",
        "surname": "new_surname",
        "email": "new_email@example.com"
    },
    "update_baho_info": {
        "name": "new_name",
        "surname": "new_surname",
        "email": "bahadir@example.com"
    }
}
```

Instead, if only 1 Body(...) parameter is set within the endpoint, then the following POST request data input MUST be used:

```json
{
        "name": "new_name",
        "surname": "new_surname",
        "email": "new_email@example.com"
}
```

and the following POST request only returns the default Pydantic data instead of the updated one in the request, which is not accepted:
```json
{
    "update_user_info": {
        "name": "new_name",
        "surname": "new_surname",
        "email": "new_email@example.com"
    }
}
```

To change it to accept only with 1 Body(...) parameter, such as following, then the POST request data input is not accepted as the one above:

```python
@app.post("/api/v1/update_profile")
async def update_profile(token: str = Header(...),
                         update_user_info: UpdateUserInfo = Body(...)):
```

Is not both POST request data architectures be allowed to be used in this specific case?

### Operating System

macOS

### Operating System Details

macOS Ventura 13.3.1 (a)

### FastAPI Version

0.95.1

### Python Version

Python 3.10.11

**Comment by bahadiraraz** [comment]
same issue

**Comment by JavierSanchezCastro** [comment]
Hi! I'm not sure I understand your requirement at all.

You can create a new class inherited from BaseModel and include your multiples body.
After in the POST method, you will have to specify to which body pertain the information you're providing.

**Example:**

```python3
from typing import Optional
from pydantic import BaseModel
from fastapi import Body, FastAPI

app = FastAPI()

class UpdateUserInfo(BaseModel):
    name: Optional[str]
    surname: Optional[str]
    email: Optional[str]

class TwoBodys(BaseModel):
    update_user_info: UpdateUserInfo = Body(...)
    #update_baho_info: UpdateUserInfo = Body(...)


@app.post("/api/v1/update_profile")
async def update_profile(body: TwoBodys):
    print(body.update_user_info)
    #print(body.update_baho_info)
    return "OK"
````

**Reply by woosal1337** [reply]
Well, the problem is you can reach the Bodies already. However, if used a single Body(...) as a parameter in an endpoint, you can not use the following type of POST request data input:


```json
{
    "update_user_info": {
        "name": "Baho",
        "surname": "Araz",
        "email": "new_email@example.com"
     }
}
```

but you must use 

```json
{
    "name": "Baho",
    "surname": "Araz",
    "email": "new_email@example.com"
}
```

With two or more Body(...) parameters in a single endpoint, however, you can use the following POST request input:

```json
{
    "update_user_info": {
        "name": "Baho",
        "surname": "Araz",
        "email": "new_email@example.com"
     },
    "update_user_info_2": {
        "name": "Baho",
        "surname": "Araz",
        "email": "new_email@example.com"
     },
}
```

**Reply by JavierSanchezCastro** [reply]
Well, If you want a single Body but specifying the "update_user_info" you can use the code I wrote like:

```python3
class SingleSpecificBody(BaseModel):
    update_user_info: UpdateUserInfo = Body(...)

@app.post("/api/v1/update_profile")
async def update_profile(body: SingleSpecificBody):
    print(body.update_user_info)
    return "OK"
```

**Reply by JavierSanchezCastro** [reply]
Did that answer help you? @woosal1337

**Reply by YuriiMotov** [reply, marked as answer]
I think that was about `Body(embed=True)`.
```python
async def update_profile(body: SingleSpecificBody = Body(embed=True)):
```
By default, if there is only one Body model, FastAPI expects data to be passed without `update_user_info` key.
```
{
    "name": "Baho",
    "surname": "Araz",
    "email": "new_email@example.com"
}
```
And `embed=True` forces FastAPI to look for the data of object inside `update_user_info` field
```
{
    "update_user_info": {
        "name": "Baho",
        "surname": "Araz",
        "email": "new_email@example.com"
     }
}
```
